AstarteDeviceSDKCSharp/Data/AstarteFailedMessageEntry.cs

```diff
@@ -54,7 +54,7 @@ public AstarteFailedMessageEntry(int qos, byte[] payload, string topic)
             Qos = qos;
             Payload = payload;
             Topic = topic;
-            absoluteExpiry = (DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()) + relativeExpiry;
+            AbsoluteExpiry = (DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()) + (relativeExpiry * 1000);
```

Comment:
The `expiry` value from AstarteInterface is in seconds, so you can use `TimeSeconds` without multiply relativeExpire by 1000
```suggestion
            AbsoluteExpiry = DateTimeOffset.UtcNow.ToUnixTimeSeconds() + relativeExpiry;
```